lib/commands/performance.js

```diff
@@ -0,0 +1,297 @@
+import { errors } from 'appium-base-driver';
+import { sleep } from 'asyncbox';
+import _ from 'lodash';
+
+let commands = {}, helpers = {}, extensions = {};
+
+commands.postDismissAlert = async function () {
+  throw new errors.NotYetImplementedError();
+};
+
+const SUPPORTED_PERFORMANCE_DATA_TYPES = {
+  'cpuinfo': 'the amount of cpu by user and kernel process - cpu information for applications on real devices and simulators',
+  'memoryinfo': 'the amount of memory used by the process - memory information for applications on real devices and simulators',
+  'batteryinfo': 'the remaining battery power - battery power information for applications on real devices and simulators',
+  'networkinfo': 'the network statistics - network rx/tx information for applications on real devices and simulators'
+};
+
+commands.getPerformanceDataTypes = function () {
+  return _.keys(SUPPORTED_PERFORMANCE_DATA_TYPES);
+};
+
+commands.getPerformanceData = async function (applicationPackageName, performanceDataType, performanceDataReadTimeout) {
+
+  if (performanceDataType === "batteryinfo")
+    return this.getBatteryInfo(performanceDataReadTimeout);
+  else if (performanceDataType === "cpuinfo")
+    return this.getCPUInfo(applicationPackageName, performanceDataReadTimeout);
+  else if (performanceDataType === "memoryinfo") 
+    return this.getMemoryInfo(applicationPackageName, performanceDataReadTimeout);
+  else if (performanceDataType === "networkinfo")
+    return this.getNetworkTrafficInfo(performanceDataReadTimeout);
+  else 
+    throw new Error('No performance data of type \'' + performanceDataType + '\' found.');
+};
+
+helpers.getCPUInfo = async function (applicationPackageName, performanceDataReadTimeout) {
```

Comment:
Same vis-a-vis parameter names. `packageName` and `dataReadTimeout` would suffice.